Drivers/Src/BatteryBalancing.c

```diff
@@ -0,0 +1,78 @@
+
+#include "LTC6811.h"
+#include "Voltage.h"
+#include "BatteryBalancing.h"
+#include "config.h"
+#include "Current.h"
+
+#define CHARGING_TOLERANCE 0
+
+static uint16_t Balancing_Init(void) {
+	uint16_t minVoltage = 0;
+	if (Current_CheckStatus(0) == 1){
+		return;//If charging, return
+	}
+	Voltage_UpdateMeasurements();
+	minVoltage = Voltage_GetModuleMillivoltage(0);
+	for (uint8_t i = 1; i < NUM_BATTERY_MODULES; i++){
+		if(Voltage_GetModuleMillivoltage(i) < minVoltage) {
+			minVoltage = Voltage_GetModuleMillivoltage(i);//Find minimum voltage
+		}
+	}
+	return minVoltage;
+}
+	
+void Balancing_Balance(cell_asic Minions[]){ 
+	uint16_t lowest = Balancing_Init(); //get lowest voltage 
+	Voltage_UpdateMeasurements();
+	for (uint8_t k = 0; k < NUM_BATTERY_MODULES; k++) {
+		uint16_t voltage = Voltage_GetModuleMillivoltage(k);//Get voltage of module
+		if (voltage > lowest + CHARGING_TOLERANCE) {//Check to see if module is greater than min
+			Balancing_SetDischarge(k, Minions);//Set discharge bit if module is too high
+		}
+		else {//Clear discharge bit of module if it reaches minimum
+			uint8_t ICIndex;
+			uint8_t MNumber;
+			Balancing_GetICNumber(k, &ICIndex, &MNumber);//Get module number and IC Board
+			Balancing_ClearDischargeBit(MNumber , 1, &Minions[ICIndex]);//Clear discharge bit
+		}
+	}
+}
+
+void Balancing_ClearDischargeBit(int Cell, uint8_t total_ic, cell_asic *ic){
```

Comment:
Can you add the comments describing each function from the header file to this file as well? (Especially if we remove some of these functions from the header file)